rfcs/generalized-docstrings.md

```diff
@@ -0,0 +1,192 @@
+# Generalize documentation comments with several syntactic categories
+
+## High-level description
+
+In the current implementation of the compiler, there is one kind of
+documentation comments. Normal comments (of the form `(* ... *)`) are dropped
+after lexing, and documentation comments (of the form `(** ... *)`) are
+collected and attached to the parsetree as attributes identified with
+`ocaml.doc`.
+
+This RFC proposes to add new kinds of documentation comments, distinguished
+syntactically from the usual documentation comments available today.
+
+The concrete syntax would be as follows:
+
+```
+(** ... *)  : a usual doc string
+(*@ ... *)  : a doc string of kind "@"
+(*! ... *)  : a doc string of kind "!"
```

Comment:
I would suggest `(**!` since `(*<char>` could easily clash with existing conventions, e.g., for many years we use `(*!` for code review in our team.

Reply:
in [qtest](https://github.com/vincent-hugot/qtest/) some custom comment prefixes are also used for inline tests: `(*$T`, `(*$Q`, etc.

Reply:
> I would suggest `(**!` since `(*<char>` could easily clash with existing conventions, e.g., for many years we use `(*!` for code review in our team.

I strongly support this idea that documentation strings "syntactic space" should be under "(**" and all other "(*X" tokens should be left for other uses.